pkg/monitoring/config.go

```diff
@@ -0,0 +1,135 @@
+package monitoring
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	relayMonitoring "github.com/smartcontractkit/chainlink-relay/pkg/monitoring"
+	"github.com/smartcontractkit/terra.go/msg"
+)
+
+// TerraConfig contains configuration for connecting to a terra client.
+type TerraConfig struct {
+	TendermintURL    string
+	NetworkName      string
+	NetworkID        string
+	ChainID          string
+	ReadTimeout      time.Duration
+	PollInterval     time.Duration
+	LinkTokenAddress sdk.AccAddress
+}
+
+var _ relayMonitoring.ChainConfig = TerraConfig{}
+
+// GetRPCEndpoint return the tendermint url of a terra client.
+func (t TerraConfig) GetRPCEndpoint() string { return t.TendermintURL }
+
+// GetNetworkName returns the network name.
+func (t TerraConfig) GetNetworkName() string { return t.NetworkName }
+
+// GetNetworkID returns the network id.
+func (t TerraConfig) GetNetworkID() string { return t.NetworkID }
+
+// GetChainID returns the chain id.
+func (t TerraConfig) GetChainID() string { return t.ChainID }
+
+// GetReadTimeout returns the max allowed duration of a request to a Terra client.
+func (t TerraConfig) GetReadTimeout() time.Duration { return t.ReadTimeout }
+
+// GetPollInterval returns the interval at which data from the chain is read.
+func (t TerraConfig) GetPollInterval() time.Duration { return t.PollInterval }
+
+// ToMapping returns a data structure expected by the Avro schema encoders.
+func (t TerraConfig) ToMapping() map[string]interface{} {
+	return map[string]interface{}{
+		"network_name": t.NetworkName,
+		"network_id":   t.NetworkID,
+		"chain_id":     t.ChainID,
+	}
+}
+
+// ParseTerraConfig extracts chain specific configuration from env vars.
+func ParseTerraConfig() (TerraConfig, error) {
+	cfg := TerraConfig{}
+
+	if err := parseEnvVars(&cfg); err != nil {
+		return cfg, err
+	}
+
+	applyDefaults(&cfg)
+
+	err := validateConfig(cfg)
```

Comment:
nit/ Would these read a little easier as methods?
```suggestion
	cfg.applyDefaults()

	err := cfg.validate()
```

Reply:
Actually yeah, that looks better. I'll do it in a seprate PR.